File: backend/internal/models/project.go
package models

import "time"

type Project struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	Name        string    `gorm:"not null" json:"name"`
	Description string    `json:"description"`
	Status      string    `gorm:"default:active" json:"status"` // active, completed, on_hold
	StartDate   string    `json:"start_date,omitempty"`
	EndDate     string    `json:"end_date,omitempty"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`

	Assignments []ProjectAssignment `gorm:"foreignKey:ProjectID" json:"assignments,omitempty"`
	Milestones  []Milestone         `gorm:"foreignKey:ProjectID" json:"milestones,omitempty"`
}

type ProjectAssignment struct {
	ID                   uint   `gorm:"primaryKey" json:"id"`
	ProjectID            uint   `gorm:"not null;uniqueIndex:idx_proj_emp_role" json:"project_id"`
	EmployeeID           uint   `gorm:"not null;uniqueIndex:idx_proj_emp_role" json:"employee_id"`
	Role                 string `gorm:"not null;uniqueIndex:idx_proj_emp_role" json:"role"` // backend, frontend, fullstack, mobile, qa, qc, ba, designer, pm, devops
	AllocationPercentage int    `gorm:"default:100" json:"allocation_percentage"` // 0-100
	StartDate            string `json:"start_date,omitempty"`
	EndDate              string `json:"end_date,omitempty"`

	Project  *Project  `gorm:"foreignKey:ProjectID" json:"project,omitempty"`
	Employee *Employee `gorm:"foreignKey:EmployeeID" json:"employee,omitempty"`
}

type Milestone struct {
	ID          uint       `gorm:"primaryKey" json:"id"`
	ProjectID   uint       `gorm:"not null" json:"project_id"`
	Title       string     `gorm:"not null" json:"title"`
	Description string     `json:"description"`
	Deadline    string     `json:"deadline,omitempty"`
	Status      string     `gorm:"default:upcoming" json:"status"` // upcoming, in_progress, completed, overdue
	CompletedAt *time.Time `json:"completed_at,omitempty"`
	CreatedAt   time.Time  `json:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"`

	Items []MilestoneItem `gorm:"foreignKey:MilestoneID" json:"items,omitempty"`
}

type MilestoneItem struct {
	ID           uint   `gorm:"primaryKey" json:"id"`
	MilestoneID  uint   `gorm:"not null" json:"milestone_id"`
	Content      string `gorm:"not null" json:"content"`
	IsCompleted  bool   `gorm:"default:false" json:"is_completed"`
	DisplayOrder int    `gorm:"default:0" json:"display_order"`
}

File: mobile/lib/features/projects/domain/entities/project.dart
import 'package:equatable/equatable.dart';

class MilestoneItem extends Equatable {
  const MilestoneItem({
    required this.id,
    required this.milestoneId,
    required this.content,
    required this.isCompleted,
    required this.displayOrder,
  });

  final int id;
  final int milestoneId;
  final String content;
  final bool isCompleted;
  final int displayOrder;

  @override
  List<Object?> get props => [id, milestoneId, content, isCompleted, displayOrder];
}

class Project extends Equatable {
  const Project({
    required this.id,
    required this.name,
    required this.status,
    required this.startDate,
    this.description,
    this.endDate,
    this.milestones = const [],
    this.members = const [],
  });

  final int id;
  final String name;
  final String status;
  final String startDate;
  final String? description;
  final String? endDate;
  final List<Milestone> milestones;
  final List<ProjectMember> members;

  @override
  List<Object?> get props => [id, name, status, startDate];
}

class ProjectMember extends Equatable {
  const ProjectMember({
    required this.id,
    required this.projectId,
    required this.employeeId,
    required this.projectRole,
    this.employeeName,
    this.joinedAt,
  });

  final int id;
  final int projectId;
  final int employeeId;
  final String projectRole;
  final String? employeeName;
  final String? joinedAt;

  @override
  List<Object?> get props => [id, projectId, employeeId, projectRole];
}

class Milestone extends Equatable {
  const Milestone({
    required this.id,
    required this.projectId,
    required this.title,
    required this.deadline,
    required this.status,
    this.description,
    this.completedAt,
    this.projectName,
    this.items = const [],
  });

  final int id;
  final int projectId;
  final String title;
  final String deadline;
  final String status;
  final String? description;
  final String? completedAt;
  final String? projectName;
  final List<MilestoneItem> items;

  bool get isOverdue {
    final deadlineDate = DateTime.tryParse(deadline);
    if (deadlineDate == null) return false;
    return deadlineDate.isBefore(DateTime.now()) && status != 'completed';
  }

  bool get isUpcoming {
    final deadlineDate = DateTime.tryParse(deadline);
    if (deadlineDate == null) return false;
    final now = DateTime.now();
    return deadlineDate.isAfter(now) &&
        deadlineDate.isBefore(now.add(const Duration(days: 7)));
  }

  @override
  List<Object?> get props => [id, projectId, title, deadline, status];
}
